include/nuttx/ioexpander/gpio.h

```diff
@@ -100,8 +100,11 @@
 
 enum gpio_pintype_e
 {
-  GPIO_INPUT_PIN = 0,
-  GPIO_OUTPUT_PIN,
+  GPIO_INPUT_PIN = 0, /* float */
+  GPIO_INPUT_PIN_PULLUP,
+  GPIO_INPUT_PIN_PULLDOWN,
+  GPIO_OUTPUT_PIN, /* push-pull */
+  GPIO_OUTPUT_PIN_OPENDRAIN,
```

Comment:
Need update drivers/ioexpander/gpio_lower_half.c too. And I don't see the related defintion in include/nuttx/ioexpander/ioexpander.h, should we add them and translate gpio call to ioexpander call?

Reply:
I extended that as well.